### DISCUSSION TITLE
Mandatory query params for same route path?

### DISCUSSION THREAD

**Discussion by iustin-gheorghe** [vuejs/router#1955] [Help and Questions] [answered]
I have this requirement where the same path behaves differently depending on the presence of a query parameter. Take the following scenario as an example: 

- `/some/path/?query=value` should render a page component 
- `/some/path/` should redirect to `/some/`.

I'm having issue declaring the first case. Registering a new router entry like `path: /some/path/:dynamicPart` is actually ignoring the query param when accessing the route and does the redirect. 

Declaring it as `path: '/some/path/:pathMatch(\\?query=)+'`, `/some/path/\\?query=:pathMatch(.+)` does not resolve the path because of the "?" char (removing the question mark makes it work but I want the dynamic part to come as a query param, not as a path param).

Tried various approaches and passed through the doc but could not find a solution. Is this even possible?

**Comment by posva** [comment, marked as answer]
the path is just the path, it doesn't contain the query, that's why your declaration with a custom regex do not work.

For this kind of scenario, using a navigation guard is the way to go:

```ts
router.beforeEach((to) => {
  if (to.name === 'my-route-to-check') {
    // ...
  }
})
```

There are a few [extra examples](https://router.vuejs.org/guide/advanced/navigation-guards.html#Navigation-Guards) in the docs. [This page](https://router.vuejs.org/guide/advanced/meta.html) is also worth checking for your case.

**Reply by iustin-gheorghe** [reply]
I see, you are right. Ended up using a Nuxt middleware. But this would work too, thanks for the answer!

**Reply by posva** [reply]
Yes, a nuxt middleware is a router guard underneath
